aesara/link/numba/linker.py

```diff
@@ -12,15 +12,26 @@ def fgraph_convert(self, fgraph, **kwargs):
         return numba_funcify(fgraph, **kwargs)
 
     def jit_compile(self, fn):
+        import numba
+
         jitted_fn = numba.njit(fn)
         return jitted_fn
 
     def create_thunk_inputs(self, storage_map):
+        from aesara.link.numba.dispatch import numba_typify
+
         thunk_inputs = []
         for n in self.fgraph.inputs:
             sinput = storage_map[n]
-            # TODO:When RandomVariable conversion is implemented
-            # do RandomState typification over here.
+            if isinstance(sinput[0], RandomState):
+                new_value = numba_typify(
+                    sinput[0], dtype=getattr(sinput[0], "dtype", None)
+                )
+                # We need to remove the reference-based connection to the
+                # original `RandomState`/shared variable's storage, because
+                # subsequent attempts to use the same shared variable within
+                # other non-JAXified graphs will have problems.
+                sinput = [new_value]
```

Comment:
We should consider changing `numba_typify` so that it returns the "storage" list instead of just the contents of said list.  It seems like that would allow us to remove the explicit `RandomState` condition in this module.